### PR TITLE
refactor!: move identity to top-level instead of sub-chart

### PR DESCRIPTION
**Pull request by aabouzaid**
### Which problem does the PR fix?

Fixes: https://github.com/camunda/camunda-platform-helm/issues/1449

### What's in this PR?

Refactor Identity and move it to the top-level chart instead of the sub-chart. Also the same for its dependencies Keycloak and Postgres.

### Checklist

Please make sure to follow our [Contributing Guide](../blob/main/CONTRIBUTING.md).

**Before opening the PR:**

- [x] In the repo's root dir, run `make go.update-golden-only`.
- [x] There is no other open [pull request](../pulls) for the same update/change.
- [x] Tests for charts are added (if needed).
- [x] In-repo [documentation](../blob/main/CONTRIBUTING.md#documentation) are updated (if needed).

**After opening the PR:**

- [x] Did you sign our CLA (Contributor License Agreement)? It will show once you open the PR.
- [x] Did all checks/tests pass in the PR?

### CODE REVIEW

**Comment by jessesimpson36** [comment]
I just tested with a combined ingress and all the components seemed to be working.

**Comment by jessesimpson36** [comment]
I'm going to try an upgrade path where I run the version before this PR, then upgrade to this PR's commit without any configuration changes, and then upgrade again after changing my values.yaml to support the 
`identity.keycloak` -> `identityKeycloak` change. If that process works out without too much broken state, then I'll approve the PR.

**Review by jessesimpson36 — APPROVED** [review]

File: charts/camunda-platform/test/integration/scenarios/lib/chart-upgrade-taskfile.yaml [hunk 1]
```diff
@@ -8,7 +8,17 @@ vars:
 tasks:
   pre:
     cmds:
-    - echo "No pre task for this test."
+    # TODO: Remove it once 8.5 cycle is done.
```
Comment (aabouzaid) [inline charts/camunda-platform/test/integration/scenarios/lib/chart-upgrade-taskfile.yaml:11]: @jessesimpson36 The needed upgrade steps are mentioned here (will be added to the official docs when all refactors are done for the parent epic).

File: charts/camunda-platform/README.md [hunk 2]
```diff
@@ -1157,90 +1149,105 @@ Please see the corresponding [release guide](../../RELEASE.md) to find out how t
...
 | `identity.externalDatabase.existingSecretPasswordKey`        | Name of an existing secret key containing the database credentials                                                                                                                                        | `nil`                       |
 
+### Identity - PostgreSQL Parameters
+
+| Name                                     | Description                                                                                                                                                                                   | Value                 |
+| ---------------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | --------------------- |
+| `postgresql`                             | configuration for the postgresql dependency chart used by Identity. See the chart documentation https://github.com/bitnami/charts/tree/master/bitnami/postgresql#parameters for more details. |                       |
```
[the remark this answers is not in the document]
Comment (aabouzaid) [reply charts/camunda-platform/README.md]: Fixed :ok_hand:

**Pull request merged**